src/main/java/wooteco/subway/admin/domain/Line.java

```diff
@@ -78,20 +89,74 @@ public void update(Line line) {
         if (line.getIntervalTime() != 0) {
             this.intervalTime = line.getIntervalTime();
         }
+        if (line.getBgColor() != null) {
+            this.bgColor = line.getBgColor();
+        }
 
         this.updatedAt = LocalDateTime.now();
     }
 
     public void addLineStation(LineStation lineStation) {
-        // TODO: 구현
+        if (Objects.isNull(lineStation.getPreStationId())) {
+            addFirst(lineStation);
+            return;
+        }
+
+        if (hasNoSuchPreStation(lineStation)) {
+            throw new NoSuchElementException("이전 역이 등록되지 않았습니다.");
+        }
+
+        Optional<LineStation> nextStation = findNextStationBy(lineStation.getPreStationId());
+        if (nextStation.isPresent()) {
+            addBetweenTwo(lineStation, nextStation.get());
+            return;
+        }
+
+        stations.add(lineStation);
+    }
+
+    private void addFirst(LineStation lineStation) {
+        stations.stream()
+                .findFirst()
+                .ifPresent(station -> station.updatePreLineStation(lineStation.getStationId()));
+        stations.add(FIRST, lineStation);
+    }
+
+    private void addBetweenTwo(LineStation lineStation, LineStation nextStation) {
+        nextStation.updatePreLineStation(lineStation.getStationId());
+        int position = stations.indexOf(nextStation);
+        stations.add(position, lineStation);
+    }
+
+    private boolean hasNoSuchPreStation(LineStation lineStation) {
+        return stations.stream()
+                .map(LineStation::getStationId)
+                .noneMatch(id -> lineStation.getPreStationId().equals(id));
+    }
+
+    private Optional<LineStation> findNextStationBy(Long stationId) {
+        return stations.stream()
+                .filter(station -> stationId.equals(station.getPreStationId()))
+                .findFirst();
     }
 
     public void removeLineStationById(Long stationId) {
-        // TODO: 구현
+        LineStation station = findStationBy(stationId);
+        findNextStationBy(stationId)
+                .ifPresent(nextStation -> nextStation.updatePreLineStation(station.getPreStationId()));
+        stations.remove(station);
+    }
+
+    private LineStation findStationBy(Long stationId) {
+        return stations.stream()
+                .filter(lineStation -> lineStation.getStationId().equals(stationId))
+                .findFirst()
+                .orElseThrow(() -> new NoSuchElementException("해당 노선에 등록되지 않은 역입니다."));
     }
 
     public List<Long> getLineStationsId() {
```

Comment:
```suggestion
    public List<Long> getLineStationIds() {
```

로 변경하는 것이 더 적절하지 않을까요?